src/EntityDiagram/ExpandedDiagram.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import {hierarchy, Tree} from "@visx/hierarchy";
+import {LinearGradient} from "@visx/gradient";
+import {Group} from "@visx/group";
+import {Text} from "@visx/text";
+import OffsetLine from "./OffsetLine"
+import {StudyData} from "./Types";
+
+interface ShadingData {
+  entityId: string
+  value: number
+  color?: string
+}
+
+interface ExpandedDiagram {
+  treeData: StudyData
+  orientation: string
+  highlightedEntityID: string
+  shadingData: ShadingData[]
+}
+
+
+
+// Todo: There MUST be a smarter way to center the text
+function CalculateDYSize(nodeLength: number) {
+  switch (nodeLength) {
+    case 1:
+      return '.33em'
+    case 2:
+      return '.80em'
+    case 3:
+      return '1.35em'
+    case 4:
+      return '1.8em'
+    case 5:
+      return '1.8em'
+  }
+}
+
+export default function MiniDiagram({treeData, orientation, highlightedEntityID, shadingData}: ExpandedDiagram) {
+  const data = hierarchy(treeData);
+
+  return (
+    <svg width="1000px" height="1000px">
+      <defs>
+        <marker
+          id="arrow"
+          viewBox="0 -5 10 10"
+          markerWidth="18"
+          markerHeight="18"
+          orient="auto"
+          fill="black"
+          style={{opacity: .7}}
+        >
+          <path d="M0,-5L10,0L0,5" />
+        </marker>
+      </defs>
+
+      {/* TODO: Background definitions: Please change this if you can think of a better way! :) */}
+      <LinearGradient vertical={false} x1={0} x2={0} fromOffset={1} id="rect-gradient-0" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.1} fromOffset={1} id="rect-gradient-1" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.2} fromOffset={1} id="rect-gradient-2" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.3} fromOffset={1} id="rect-gradient-3" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.4} fromOffset={1} id="rect-gradient-4" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.5} fromOffset={1} id="rect-gradient-5" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.6} fromOffset={1} id="rect-gradient-6" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.7} fromOffset={1} id="rect-gradient-7" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.8} fromOffset={1} id="rect-gradient-8" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={.9} fromOffset={1} id="rect-gradient-9" from="#e4c8c8" to="white" />
+      <LinearGradient vertical={false} x1={0} x2={1} fromOffset={1} id="rect-gradient-10" from="#e4c8c8" to="white" />
```

Comment:
I'm not sure what this is for exactly, but this bit of code can be changed to the following:

```suggestion
      {Array(11).fill(null).map((_, index) =>
        <LinearGradient vertical={false} x1={0} x2={index * 0.1} fromOffset={1} id={`rect-gradient-${index}`} from="#e4c8c8" to="white"/>
      }
```

Reply:
Oh awesome! This is for the background gradients in the nodes. Great solution 👏